fabric-biomes-v1/src/main/java/net/fabricmc/fabric/api/biomes/v1/BiomeEdgeGenPredicate.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.biomes.v1;
+
+import java.util.Collection;
+
+import com.google.common.collect.Lists;
+
+import net.minecraft.world.biome.Biome;
+import net.minecraft.world.biome.layer.LayerRandomnessSource;
+
+/**
+ * A predicate of a biome array and randomness source.
+ */
+public interface BiomeEdgeGenPredicate {
+	/**
+	 * @param biomes the biomes neighboring generation
+	 * @param rand the world gen randomness source
+	 * @return whether the conditions for generation are met
+	 */
+	boolean isValidToGenerate(Collection<Biome> biomes, LayerRandomnessSource rand);
+
+	/**
+	 * @return a {@link BiomeEdgeGenPredicate} which returns true if any of the given biomes border the parent biome.
+	 */
+	static BiomeEdgeGenPredicate createBiomePredicate(Biome biome, Biome...otherBiomes) {
```

Comment:
```suggestion
	static BiomeEdgeGenPredicate matchAnyBiome(Biome biome, Biome...otherBiomes) {
```

Reply:
`create` is the standard factory method name tho